### PR TITLE
[COM-277] Implementation of Circular Progress component

### PR DESCRIPTION
**Pull request by Angele91**
## Please delete checklist items and sections that are not relevant

**Checklist**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)

https://impulsumvc.atlassian.net/browse/COM-277

**Storybook Link** (Link to a local storybook url to review new functionality)

http://localhost:8080/?path=/story/components-primitives-circularprogress--all

### CODE REVIEW

File: packages/components/src/components/primitives/CircularProgress/CircularProgress.test.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { renderWithProviders } from 'tests/renderWithProviders';
+import { CircularProgress } from '.';
+
+describe('CircularProgress', () => {
+  test('should render', () => {
+    renderWithProviders(<CircularProgress value={10} />);
```
Comment (guilean) [inline packages/components/src/components/primitives/CircularProgress/CircularProgress.test.tsx:8]: Should we expect an assertion? @Angele91

Comment (Angele91) [reply packages/components/src/components/primitives/CircularProgress/CircularProgress.test.tsx:8]: fixed: added an assertion for the render test

File: packages/components/src/components/primitives/CircularProgress/CircularProgress.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  return (
+    <ChakraCircularProgress
+      trackColor={trackColor as string}
+      color={filledTrackColor as string}
+      data-testid="cmpsr.circular-progress"
+      {...props}
+    >
+      {showLabel && (
+        <CircularProgressLabel color={color as string}>
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/CircularProgress/CircularProgress.tsx:27]: this should use our `Text` component with the appropriate variant

File: packages/components/src/components/primitives/CircularProgress/CircularProgress.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import { CircularProgress } from './CircularProgress';
+import { HStack, Text, VStack } from '@chakra-ui/react';
+
+export default {
+  component: CircularProgress,
+  title: 'Components/Primitives/CircularProgress',
+  argTypes: {
+    size: {
+      control: { type: 'number' },
+    },
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/CircularProgress/CircularProgress.stories.tsx:12]: this is defined as a `number` in here but the value in the [arguments](https://github.com/cmpsr/composer/pull/306/files#diff-59fcda40313833c8ba0b5580caa163e1e454256230dfc80cacb53ff2d6094636R44) is a string.

Also the behaviour is errant:
Screenshot 2022-01-18 at 18 38 33
Screenshot 2022-01-18 at 18 38 42

File: packages/components/src/components/primitives/CircularProgress/types.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { CircularProgressProps as ChakraCircularProgressProps } from '@chakra-ui/react';
+
+
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/CircularProgress/types.ts:3]: can you confirm do you have the prettier extension installed and active?, I've been seeing styling issues in your PRs that should be automatically fixed by prettier

File: packages/components/src/components/primitives/CircularProgress/CircularProgress.tsx (hunk 2 above)
Comment (Angele91) [reply packages/components/src/components/primitives/CircularProgress/CircularProgress.tsx]: done!

File: packages/components/src/components/primitives/CircularProgress/CircularProgress.stories.tsx (hunk 3 above)
Comment (Angele91) [reply packages/components/src/components/primitives/CircularProgress/CircularProgress.stories.tsx]: done!

File: packages/components/src/components/primitives/CircularProgress/types.ts (hunk 4 above)
Comment (Angele91) [reply packages/components/src/components/primitives/CircularProgress/types.ts]: i have fixed the formatting errors.
i have the extension, but the _format on save_ option was disabled!

File: packages/components/src/theme/ComposerTheme/Components/CircularProgress.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+      position: 'absolute',
+      transform: 'translate(-50%, -50%)',
+    },
+    track: {
+      backgroundColor: 'background-static',
+    },
+    filledTrack: {
+      backgroundColor: 'primary-default',
+    },
+  },
```
Comment (tkanzakic) [inline packages/components/src/theme/ComposerTheme/Components/CircularProgress.ts:20]: nit: can we set a default size of `5rem` as in figma?

**Review by tkanzakic — APPROVED** [review]
small comment and feel free to merge

**Review by guilean — APPROVED** [review]
LGTM 👍🏻 Solve conflicts and should be ready to merge

**Pull request merged**